src/sentry/static/sentry/app/components/search/sources/apiSource.tsx

```diff
@@ -1,26 +1,59 @@
 import React from 'react';
-import {withRouter} from 'react-router';
+import {withRouter, WithRouterProps} from 'react-router';
 import * as Sentry from '@sentry/react';
 import debounce from 'lodash/debounce';
 import flatten from 'lodash/flatten';
-import PropTypes from 'prop-types';
 
 import {Client} from 'app/api';
 import {t} from 'app/locale';
-import SentryTypes from 'app/sentryTypes';
-import {createFuzzySearch} from 'app/utils/createFuzzySearch';
+import {
+  Group,
+  IntegrationProvider,
+  Member,
+  Organization,
+  PluginWithProjectList,
+  Project,
+  SentryApp,
+  Team,
+} from 'app/types';
+import {Event} from 'app/types/event';
+import {defined} from 'app/utils';
+import {createFuzzySearch, isResultWithMatches} from 'app/utils/createFuzzySearch';
 import {singleLineRenderer as markedSingleLine} from 'app/utils/marked';
 import withLatestContext from 'app/utils/withLatestContext';
 import {documentIntegrationList} from 'app/views/organizationIntegrations/constants';
 
+import {ChildProps, Result, ResultItem} from './types';
+
+// Response from ShortIdLookupEndpoint
+type ShortIdResponse = {
+  organizationSlug: string;
+  projectSlug: string;
+  groupId: string;
+  group: Group;
+  shortId: string;
+};
+
+// Response from EventIdLookupEndpoint
+type EventIdResponse = {
+  organizationSlug: string;
+  projectSlug: string;
+  groupId: string;
+  eventId: string;
+  event: Event;
+};
+
 // event ids must have string length of 32
-const shouldSearchEventIds = query => typeof query === 'string' && query.length === 32;
+const shouldSearchEventIds = (query?: string) =>
+  typeof query === 'string' && query.length === 32;
```

Comment:
Shouldn't actually need this typecheck anymore now that we've locked this to being a string. But always good to be safe probably.

Reply:
I didn't want to change the runtime code too much at this point.